src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -524,7 +549,115 @@ async def _update_repo_using_branch_head(repo: GitRepo) -> Optional[RepoStatus]:
     )
 
 
-async def _check_for_changes_in_repositories(
+async def _git_sha_of_tag(repo_path: str, tag: str) -> str:
+    cmd = ["git", "rev-list", "-n", "1", tag]
+    try:
+        data = await exec_command_async(cmd, repo_path)
+    except CmdLineError as e:
+        raise RuntimeError(
+            " ".join(cmd), "The command was invalid and the cmd call failed."
+        ) from e
+    if not data:
+        raise RuntimeError(
+            "Tag", tag, " does not exist on repo", repo_path, ". Aborting!"
+        )
+    return data.strip()
+
+
+async def _get_tags_associated_to_sha(repo_path: str, sha: str) -> list[str]:
+    cmd = ["git", "tag", "--points-at", sha]
+    try:
+        data = await exec_command_async(cmd, repo_path)
+    except CmdLineError as e:
+        raise RuntimeError(
+            " ".join(cmd), "The command was invalid and the cmd call failed."
+        ) from e
+    if not data:
+        return []
+    return data.split()
+
+
+async def _get_repos_latest_tags(
+    repos: list[GitRepo],
+) -> list[Any]:
+    each_repo_latest_tags = []
+    for repo in repos:
+        if not repo.tags:
+            continue
+        current_regexp = repo.tags
+        any_matching_tag = (
+            await _git_get_latest_matching_tag(  # This returns only one tag
+                repo.directory, repo.tags
+            )
+        )
+        if any_matching_tag:
+            sha_of_tag = await _git_sha_of_tag(repo.directory, any_matching_tag)
+            all_tags_of_sha = await _get_tags_associated_to_sha(
+                repo.directory, sha_of_tag
+            )
+            # Retain only regexp-matching tags
+            all_matching_tags_of_sha = [
+                tag for tag in all_tags_of_sha if re.search(current_regexp, tag) != None
+            ]
+            each_repo_latest_tags.append((repo.repo_id, all_matching_tags_of_sha))
```

Comment:
you can also use early return here 
```suggestion
        if not any_matching_tag:
            continue
        sha_of_tag = await _git_sha_of_tag(repo.directory, any_matching_tag)
        all_tags_of_sha = await _get_tags_associated_to_sha(
            repo.directory, sha_of_tag
        )
        # Retain only regexp-matching tags
        all_matching_tags_of_sha = [
            tag for tag in all_tags_of_sha if re.search(current_regexp, tag) != None
        ]
        each_repo_latest_tags.append((repo.repo_id, all_matching_tags_of_sha))
```

Reply:
adressing this, thx